app/logical/sources/strategies/hentai_foundry.rb

```diff
@@ -0,0 +1,114 @@
+# Image URLs
+#
+# * https://pictures.hentai-foundry.com/a/Afrobull/795025/Afrobull-795025-kuroeda.png
+#
+# Page URLs
+#
+# * https://www.hentai-foundry.com/pictures/user/Afrobull/795025/kuroeda
+# * https://www.hentai-foundry.com/pictures/user/Afrobull/795025o
+#
+# Preview URLs
+#
+# * https://thumbs.hentai-foundry.com/thumb.php?pid=795025&size=350
+#
+# Profile URLs
+#
+# * https://www.hentai-foundry.com/user/kajinman/profile
+# * https://www.hentai-foundry.com/pictures/user/kajinman
+# * https://www.hentai-foundry.com/pictures/user/kajinman/scraps
+# * https://www.hentai-foundry.com/user/J-likes-to-draw/profile
+
+module Sources
+  module Strategies
+    class HentaiFoundry < Base
+      BASE_URL =    %r!\Ahttps?://(?:www\.)?hentai-foundry\.com!i
+      PAGE_URL =    %r!#{BASE_URL}/pictures/user/(?<artist_name>[\w-]+)/(?<illust_id>\d+)(?:/[\w-]*)?(\?[\w=]*)?\z!i
+      PROFILE_URL = %r!#{BASE_URL}/(?:pictures/)?user/(?<artist_name>[\w-]+)(?:/[a-z]*)?\z!i
+      IMAGE_URL =   %r!\Ahttps?://pictures\.hentai-foundry\.com/\w/(?<artist_name>[\w-]+)/(?<illust_id>\d+)/[\w-]+\.\w+\z!i
+
+      def domains
+        ["hentai-foundry.com"]
+      end
+
+      def site_name
+        "Hentai Foundry"
+      end
+
+      def image_urls
+        image = page&.search("#picBox img")
+
+        return [] unless image
+
+        image.to_a.map { |img| "https:#{img["src"]}" }
+      end
+
+      def preview_urls
+        image_urls.map do |img|
+          "https://thumbs.hentai-foundry.com/thumb.php?pid=#{img[IMAGE_URL, :illust_id]}&size=250"
+        end
+      end
+
+      def page_url
+        return nil if illust_id.blank? || artist_name.blank?
+        "https://www.hentai-foundry.com/pictures/user/#{artist_name}/#{illust_id}"
+      end
+
+      def page
+        return nil if page_url.blank?
+
+        doc = Cache.get("hentai-foundry:#{page_url}", 1.minute) do
+          HTTParty.get("#{page_url}?enterAgree=1").body
+        end
+
+        Nokogiri::HTML(doc)
+      end
+
+      def tags
+        tags = page&.search(".boxbody [rel='tag']")&.to_a
+
+        return [] unless tags
```

Comment:
nitpick: this seems a bit strange to me:
* you could get rid of all this safe navigation (`&.`) stuff if you did `return [] if page.blank?` at the start of the method
  * then you wouldn't need this `return [] unless tags` because it's just there in case `tags` is nil which should only happen if `page` is nil
* the `.to_a` seems unnecessary


There seems to be nothing wrong with using those but I didn't actually know Nokogiri had a `#search` method - I was always using the `#css` (seems to do the same) and `#at_css` (only returns one result) from [its tutorial](https://nokogiri.org/tutorials/searching_a_xml_html_document.html)

Reply:
> because it's just there in case tags is nil which should only happen if page is nil

Pictures can have no tags on hentai-foundry. Or am I misunderstanding you?

Reply:
Well, yes, but if `#search` can't find any tags, it just returns an empty Nodeset / array / whatever and `[].map { ... } == []` so there's no need to return early

Reply:
Hm, not sure why i did it like this, but right now doing 
```ruby

      def tags
        tags = page&.search(".boxbody [rel='tag']") || []

        tags.map do |tag|
          [tag.text, "https://www.hentai-foundry.com" + tag.attr("href")]
        end
      end
```

Still works. That's fine too right?

I took the "or empty array" from https://github.com/danbooru/danbooru/blob/master/app/logical/sources/strategies/nijie.rb#L121

Reply:
If you're wondering about why I use `.to_a` in certain places, this is a trick to convert `nil` to an empty array, which lets us avoid extra conditionals in some places.